packages/fs/fs.walk/src/settings.ts

```diff
@@ -0,0 +1,56 @@
+import * as fsScandir from '@nodelib/fs.scandir';
+
+import { Entry, ScandirError } from './types/index';
+
+export type EntryFilterFunction = (entry: Entry) => boolean;
+export type ErrorFilterFunction = (error: ScandirError) => boolean;
+
+export interface Options {
+	concurrency?: number;
+	entryFilter?: EntryFilterFunction;
```

Comment:
We have to divide this filter into two: entry and directory. If we do not do this, we will not be able to take only files, as the directories will not be read.